packages/AI/VectorDupe/src/duplicateRecordDetector.ts

```diff
@@ -369,8 +381,8 @@ export class DuplicateRecordDetector extends VectorBase {
                     //merge
                     let mergeParams: RecordMergeRequest = new RecordMergeRequest();
                     mergeParams.EntityName = entityDocument.Entity;
-                    mergeParams.SurvivingRecordPrimaryKeyValues = dupeResult.RecordPrimaryKeys.PrimaryKeyValues;
-                    mergeParams.RecordsToMerge = [dupe.PrimaryKeyValues];
+                    mergeParams.SurvivingRecordKeyValuePairs = dupeResult.RecordPrimaryKeys.KeyValuePairs;
```

Comment:
@JS-BC  why is this still KeyValuePairs? You should have gone into the `RecordMergeRequest` class and cleaned it up to use `CompositeKey`. When we are doing core work like this you have to go down the stack as much as needed to you don't have old style remnant laying around for years into the future :)